src/Grids/convenience_constructors.jl

```diff
@@ -0,0 +1,680 @@
+end
+
+"""
+    DefaultRectangleXYMesh(
+        ::Type{<:AbstractFloat}; # defaults to Float64
+        x_min::Real,
+        x_max::Real,
+        y_min::Real,
+        y_max::Real,
+        periodic_x::Bool,
+        periodic_y::Bool,
+    )
+
+A convenience constructor, which builds a
+`RectilinearMesh` with a rectangular domain
+composed of interval domains.
+"""
+DefaultRectangleXYMesh(; kwargs...) = DefaultRectangleXYMesh(Float64; kwargs...)
+function DefaultRectangleXYMesh(
+    ::Type{FT};
+    x_min::Real,
+    x_max::Real,
+    y_min::Real,
+    y_max::Real,
+    x_elem::Integer,
+    y_elem::Integer,
+    periodic_x::Bool,
+    periodic_y::Bool,
+) where {FT <: AbstractFloat}
+    x1boundary = (:east, :west)
+    x2boundary = (:south, :north)
+    domain = Domains.RectangleDomain(
+        Domains.IntervalDomain(
+            Geometry.XPoint{FT}(x_min),
+            Geometry.XPoint{FT}(x_max);
+            periodic = periodic_x,
+            boundary_names = x1boundary,
+        ),
+        Domains.IntervalDomain(
+            Geometry.YPoint{FT}(y_min),
+            Geometry.YPoint{FT}(y_max);
+            periodic = periodic_y,
+            boundary_names = x2boundary,
+        ),
+    )
+    return Meshes.RectilinearMesh(domain, x_elem, y_elem)
+end
+
+#####
+##### Grids
+#####
+
+"""
+    ExtrudedCubedSphereGrid(
+        ::Type{<:AbstractFloat}; # defaults to Float64
+        z_elem::Integer,
+        z_min::Real,
+        z_max::Real,
+        radius::Real,
+        h_elem::Integer,
+        n_quad_points::Integer,
+        device::ClimaComms.AbstractDevice = ClimaComms.device(),
+        context::ClimaComms.AbstractCommsContext = ClimaComms.context(device),
+        stretch::Meshes.StretchingRule = Meshes.Uniform(),
+        hypsography::HypsographyAdaption = Flat(),
+        global_geometry::Geometry.AbstractGlobalGeometry = Geometry.ShallowSphericalGlobalGeometry(radius),
+        quad::Quadratures.QuadratureStyle = Quadratures.GLL{n_quad_points}(),
+        h_mesh = Meshes.EquiangularCubedSphere(Domains.SphereDomain{FT}(radius), h_elem),
+        h_topology::Topologies.AbstractDistributedTopology = Topologies.Topology2D(context, h_mesh),
+        horizontal_layout_type = DataLayouts.IJFH,
+        z_mesh::Meshes.IntervalMesh = DefaultZMesh(FT; z_min, z_max, z_elem, stretch),
+        enable_bubble::Bool = false
+    )
+
+A convenience constructor, which builds an
+`ExtrudedFiniteDifferenceGrid`, given:
+
+ - `FT` the float type (defaults to `Float64`) [`Float32`, `Float64`]
+ - `z_elem` the number of z-points
+ - `z_min` the domain minimum along the z-direction.
+ - `z_max` the domain maximum along the z-direction.
+ - `radius` the radius of the cubed sphere
+ - `h_elem` the number of elements per side of every panel (6 panels in total)
+ - `n_quad_points` the number of quadrature points
+ - `device` the [`ClimaComms.device`](@ref)
+ - `context` the [`ClimaComms.context`](@ref)
```

Comment:
We should add links to the documentation for ClimaComms here

Reply:
Something wasn't working with that, we can iterate on this, though.